### PR TITLE
Fixed #596 - Handle SQL 'attach' table incompatibility in CBL 1.1

### LINKED ISSUE #596: Handle SQL 'attach' table incompatibility in CBL 1.1
Per @snej's guidance:
```
detect whether a database was created on Obj-C-1.1 and if so recreate the ‘attach’ table.

When opening a database, check whether the ‘attach’ table exists. If not:
Create the ‘attach’ table.
Iterate over every revision containing attachments (rows in ‘revs’ whose ‘noattach' column is not 1.)
Parse the revision’s JSON and get the ‘_attachments’ property.
For every item in that map, create a row in the ‘attach’ table. (It may be possible to reuse or at least copy existing Database code, since this is the same thing that happens when adding a revision to the database.)
```

### PR DESCRIPTION
**Pull request by hideki**
- Check dbVersion. If >= 200, current version can not support
- If dbVersion >= 100, Check if attachments table exists. If not, create attachments table, and iterates revs table to populate attachments table
- Set current database version 20.  We should use the number less than 101 and larger than previous released version

### CODE REVIEW

File: src/main/java/com/couchbase/lite/Database.java [hunk 1]
```diff
@@ -1210,6 +1211,111 @@ public synchronized boolean open() {
...
+                    }
+                    database.close();
+                    return false;
+                } finally {
+                    if (cursor2 != null) {
+                        cursor2.close();
+                    }
+                }
+            }
+            dbVersion = 20;
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: Do you need to change the user_version to 20 also?

File: src/main/java/com/couchbase/lite/Database.java [hunk 2]
```diff
@@ -1210,6 +1211,111 @@ public synchronized boolean open() {
...
+                        cursor2.close();
+                    }
+                    database.close();
+                    return false;
+                } finally {
+                    if (cursor2 != null) {
+                        cursor2.close();
+                    }
+                }
+            }
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: I could be faster if you are using transaction.

File: src/main/java/com/couchbase/lite/Database.java (hunk 1 above)
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: I see. You have "PRAGMA user_version = 20" in the upgrade sql.

File: src/main/java/com/couchbase/lite/Database.java (hunk 2 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/Database.java]: We will do it after 1.1 release. 
Here is ticket for it https://github.com/couchbase/couchbase-lite-java-core/issues/607

File: src/main/java/com/couchbase/lite/Database.java (hunk 1 above)
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: I think if there is an error while migrating, we should rollback everything (including the created attachments table and schema version).

**Pull request merged**